client/components/pages/HomePage.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { Link } from 'react-router';
+import Divider from 'material-ui/Divider';
+import Nav from '../layouts/Nav.jsx';
+import Sidebar from '../layouts/Sidebar.jsx';
+import Searchbar from '../forms/Searchbar.jsx';
+import Pagination from '../elements/Pagination.jsx';
+import * as userActions from '../../actions/userActions';
+import * as searchActions from '../../actions/searchActions';
+import * as documentActions from '../../actions/documentActions';
+
+class HomePage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      documents: [...props.documents],
+      pageData: Object.assign({}, props.pageData),
+      documentsLoaded: false
+    };
+    this.nextPage = this.nextPage.bind(this);
+    this.prevPage = this.prevPage.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.actions.getDocuments();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.state.pageData !== nextProps.pageData) {
+      this.setState({
+        documents: [...nextProps.documents],
+        pageData: Object.assign({}, nextProps.pageData),
+        documentsLoaded: true
+      });
+    }
+  }
+
+  nextPage() {
+    if (this.state.documents.length < 9) {
+      return;
+    }
+    if (this.state.pageData.query) {
+      return this.props.actions.searchDocuments(this.state.pageData.query,
+        this.state.pageData.offset + 9);
+    }
```

Comment:
I'm a bit uncomfortable seeing `magic numbers` here even though I can rightly guess what they do.
Also I wonder if this is the best place for this manipulation? Perhaps a utility `pagination` function might be a good idea?